lib/ui/survey_question/survey_question_views/survey_question_view.dart

```diff
@@ -0,0 +1,119 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:go_router/go_router.dart';
+import 'package:survey_flutter_ic/gen/assets.gen.dart';
+import 'package:survey_flutter_ic/ui/survey_question/survey_question_component_id.dart';
+import 'package:survey_flutter_ic/ui/survey_question/survey_questions_screen.dart';
+import 'package:survey_flutter_ic/ui/survey_question/ui_models/survey_question_ui_model.dart';
+import 'package:survey_flutter_ic/utils/context_ext.dart';
+import 'package:survey_flutter_ic/utils/custom_app_bar.dart';
+import 'package:survey_flutter_ic/utils/route_path.dart';
+
+class SurveyQuestionView extends StatelessWidget {
+  final SurveyQuestionUIModel uiModel;
+  final Widget child;
+  final Function() onNextQuestion;
+  final Function() onSubmit;
+
+  const SurveyQuestionView({
+    super.key,
+    required this.uiModel,
+    required this.child,
+    required this.onNextQuestion,
+    required this.onSubmit,
+  });
+
+  AppBar _appBar(BuildContext context) => CustomAppBar.closeButton(
+        context: context,
+        onPressed: () => context.pushReplacementNamed(
+          RoutePath.home.name,
+        ),
+      );
+
+  Widget _background(BuildContext context) => SizedBox(
+        width: context.screenSize.width,
+        height: context.screenSize.height,
+        child: Image(
+          image: Assets.images.bgLoginOverlay.image().image,
+          fit: BoxFit.cover,
+        ),
+      );
+
+  Widget _mainBody(BuildContext context) => Container(
+        width: context.screenSize.width,
+        padding: const EdgeInsets.all(20),
+        child: Column(
+          mainAxisAlignment: MainAxisAlignment.start,
+          crossAxisAlignment: CrossAxisAlignment.start,
+          children: [
+            _questionNumber(context),
+            const SizedBox(
+              height: 16,
+            ),
+            _questionTitle(context),
+            Expanded(child: child),
+          ],
+        ),
+      );
+
+  Widget _questionNumber(BuildContext context) =>
+      Consumer(builder: (_, ref, __) {
+        final state = ref.watch(surveyQuestionsViewModelProvider);
+        return state.maybeWhen(
+          orElse: () => const SizedBox.shrink(),
+          success: (uiModel) => Text(
+            uiModel.question.currentIndexPerTotal,
+            style: context.textTheme.bodyMedium,
+          ),
```

Comment:
I suggest that we should put `orElse` in the bottom of the statement
```suggestion
          success: (uiModel) => Text(
            uiModel.question.currentIndexPerTotal,
            style: context.textTheme.bodyMedium,
          ),
          orElse: () => const SizedBox.shrink(),
```